src/history.cpp

```diff
@@ -1453,7 +1453,10 @@ bool history_t::save_internal_via_rewrite()
                 if (wstat(new_name, &sbuf) >= 0)
                 {
                    /* Success */
-                    fchown(out_fd, sbuf.st_uid, sbuf.st_gid);
+                    if (0 > fchown(out_fd, sbuf.st_uid, sbuf.st_gid))
+                    {
+                        debug(2, L"Error when changing ownership of history file");
```

Comment:
Sadly I don't think you can use safe_strerror() because it doesn't return a wstring. This is something else I think the project should handle using a better abstraction (e.g., a "debug_errno()" function). For now just print the decimal errno value so that anyone debugging a problem with this logic has some idea why it failed.